drivers/interrupt_controller/swerv_pic.c

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2019 Western Digital Corporation or its affiliates
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief SweRV EH1 PIC driver
+ */
+
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <init.h>
+#include <sw_isr_table.h>
+
+#define SWERV_PIC_MAX_NUM	CONFIG_NUM_IRQS
+#define SWERV_PIC_MAX_ID	(SWERV_PIC_MAX_NUM + RISCV_MAX_GENERIC_IRQ)
+#define SWERV_PIC_MAX_PRIO		16
+
+#define SWERV_PIC_mpiccfg		0x3000
+#define SWERV_PIC_meipl(s)		(0x0 + (s)*4)
+#define SWERV_PIC_meip(x)		(0x1000 + (x)*4)
+#define SWERV_PIC_meie(s)		(0x2000 + (s)*4)
+#define SWERV_PIC_meigwctrl(s)		(0x4000 + (s)*4)
+#define SWERV_PIC_meigwclr(s)		(0x5000 + (s)*4)
+
+#define SWERV_PIC_meivt			"0xBC8"
+#define SWERV_PIC_meipt			"0xBC9"
+#define SWERV_PIC_meicpct		"0xBCA"
+#define SWERV_PIC_meicidpl		"0xBCB"
+#define SWERV_PIC_meicurpl		"0xBCC"
+#define SWERV_PIC_meihap		"0xFC8"
+
+#define swerv_piccsr(csr) SWERV_PIC_##csr
+
+#define swerv_pic_readcsr(csr, value) \
+	volatile("csrr %0, "swerv_piccsr(csr) : "=r" (value))
+#define swerv_pic_writecsr(csr, value) \
+	volatile("csrw "swerv_piccsr(csr)", %0" :: "rK" (value))
+
+static int save_irq;
+
+static u32_t swerv_pic_read(u32_t reg)
+{
+	return *(volatile u32_t *)(DT_INST_0_SWERV_PIC_BASE_ADDRESS + reg);
+}
+
+static void swerv_pic_write(u32_t reg, u32_t val)
+{
+	*(volatile u32_t *)(DT_INST_0_SWERV_PIC_BASE_ADDRESS + reg) = val;
+}
+
+void swerv_pic_irq_enable(u32_t irq)
+{
+	u32_t key;
+
+	if ((irq >= SWERV_PIC_MAX_ID) || (irq < RISCV_MAX_GENERIC_IRQ)) {
+		return;
+	}
+
+	key = irq_lock();
+	swerv_pic_write(SWERV_PIC_meie(irq - RISCV_MAX_GENERIC_IRQ), 1);
+	irq_unlock(key);
+}
+
+void swerv_pic_irq_disable(u32_t irq)
+{
+	u32_t key;
+
+	if ((irq >= SWERV_PIC_MAX_ID) || (irq < RISCV_MAX_GENERIC_IRQ)) {
+		return;
+	}
+
+	key = irq_lock();
+	swerv_pic_write(SWERV_PIC_meie(irq - RISCV_MAX_GENERIC_IRQ), 0);
+	irq_unlock(key);
+}
+
+int swerv_pic_irq_is_enabled(u32_t irq)
+{
+	if ((irq >= SWERV_PIC_MAX_ID) || (irq < RISCV_MAX_GENERIC_IRQ)) {
+		return -1;
+	}
+
+	return swerv_pic_read(SWERV_PIC_meie(irq - RISCV_MAX_GENERIC_IRQ))
+	  & 0x1;
+}
+
+void swerv_pic_set_priority(u32_t irq, u32_t priority)
+{
+	u32_t key;
+
+	if (irq <= RISCV_MAX_GENERIC_IRQ)
```

Comment:
please add braces here https://docs.zephyrproject.org/latest/contribute/index.html#coding-style

Reply:
Thanks. Looks like both checkpatch and I missed that one. Fixed, rebased and pushed